From: Dmitry Shachnev <mitya57@debian.org>
Date: Sat, 16 Jul 2022 10:42:44 +0800
Subject: add -latomic to dependencies of base component on mipsel

This is needed to fix build of qwebengine_convert_dict, which uses the
generated convert_dict.pri to get the list of libraries.

qmake adds its own -latomic, but it comes before the list of static
libraries, so it does not help. We need -latomic after that list.

The error was:

/usr/bin/ld.bfd: <<BUILDDIR>>/src/core/release/obj/base/base/base_jumbo_17.o: undefined reference to symbol '__atomic_load_8@@LIBATOMIC_1.0'
/usr/bin/ld.bfd: /usr/lib/gcc/mipsel-linux-gnu/9/libatomic.so: error adding symbols: DSO missing from command line
Forwarded: no
Last-Update: 2021-03-08
---
 src/3rdparty/chromium/base/BUILD.gn | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/3rdparty/chromium/base/BUILD.gn b/src/3rdparty/chromium/base/BUILD.gn
index 9b3aa20..57ac485 100644
--- a/src/3rdparty/chromium/base/BUILD.gn
+++ b/src/3rdparty/chromium/base/BUILD.gn
@@ -1309,7 +1309,7 @@ jumbo_component("base") {
   # building inside the cros_sdk environment - use host_toolchain as a
   # more robust check for this.
   if (!use_sysroot &&
-      (is_android || ((is_linux || is_chromeos) && !is_chromecast && !use_qt)) &&
+      (is_android || ((is_linux || is_chromeos) && !is_chromecast && !use_qt) || current_cpu == "mipsel") &&
       host_toolchain != "//build/toolchain/cros:host") {
     libs += [ "atomic" ]
   }
